app/controllers/search.js

```diff
@@ -1,16 +1,19 @@
 import Controller from '@ember/controller';
 import { computed } from '@ember/object';
+import { alias, bool, readOnly } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
 
 import PaginationMixin from '../mixins/pagination';
 
 export default Controller.extend(PaginationMixin, {
+    search: service(),
     queryParams: ['q', 'page', 'per_page', 'sort'],
-    q: null,
+    q: alias('search.q'),
```

Comment:
am I understanding correctly that we're now redirecting the `q` query param to the `q` property of the `search` service?

Reply:
yes. It allows to sync input value with query string in the url.